openssl/src/ssl/mod.rs

```diff
@@ -3484,6 +3484,47 @@ impl SslRef {
             }
         }
     }
+
+    /// Returns the NID of the negotiated group used for the handshake key
+    /// exchange process.
+    /// For TLSv1.3 connections this typically reflects the state of the
+    /// current connection, though in the case of PSK-only resumption, the
+    /// returned value will be from a previous connection.
+    /// For earlier TLS versions, when a session has been resumed, it always
+    /// reflects the group used for key exchange during the initial handshake
+    /// (otherwise it is from the current, non-resumption, connection).
+    /// This can be called by either client or server.
+    /// If the NID for the shared group is unknown then the value is set to the
+    /// bitwise OR of TLSEXT_nid_unknown (0x1000000) and the id of the group.
+    #[corresponds(SSL_get_negotiated_group)]
+    #[cfg(ossl300)]
+    pub fn negotiated_group(&self) -> Result<c_int, ErrorStack> {
+        unsafe { cvt(ffi::SSL_get_negotiated_group(self.as_ptr())) }
+    }
+
+    /// Return retrieve the TLS group name associated with a given TLS
+    /// group ID, as registered via built-in or external providers and as
+    /// returned by a call to SSL_get1_groups() or SSL_get_shared_group().
+    ///
+    /// If non-NULL, SSL_group_to_name() returns the TLS group name
+    /// corresponding to the given id as a NUL-terminated string.
+    /// If SSL_group_to_name() returns NULL, an error occurred; possibly no
+    /// corresponding tlsname was registered during provider initialisation.
+    ///
+    /// Note that the return value is valid only during the lifetime of the
+    /// SSL object ssl.
+    #[corresponds(SSL_group_to_name)]
+    #[cfg(ossl300)]
+    pub fn group_to_name(&self, id: c_int) -> Result<&str, ErrorStack> {
+        unsafe {
+            match cvt_p_const(ffi::SSL_group_to_name(self.as_ptr(), id)) {
+                Ok(constp) => Ok(CStr::from_ptr(constp)
+                    .to_str()
+                    .expect("Invalid UTF8 in input")),
```

Comment:
Having `.expect()` here seems wrong to me, but I am not sure what is the best way to address this, consistently with best practices in `rust-openssl`.

Reply:
I think this is fine - the set of names that can be returned should be a static set from within OpenSSL rather than being provided by the peer.

Reply:
The string could be coming from a 3rd party provider, but in the OpenSSL threat model, one should trust the providers they load.

Alternatively is there a way to fold this Rust error within the returned `ErrorStack` type?